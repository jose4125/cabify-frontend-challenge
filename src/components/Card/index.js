import React from 'react';

import cabifyLogo from '../../images/cabify-logo.svg';

const Card = ({ info }) => {
  const {
    fullName,
    jobDescription,
    prefix,
    phoneNumber,
    email,
    website,
    address
  } = info;

  return (
    <article className="businessCard col col6">
      <figure className="businessCard-badge">
        <a className="businessCard-badge-logo" href="http://www.cabify.com">
          <img src={cabifyLogo} alt="Cabify" />
        </a>
      </figure>
      <h1 className="title-main">Request your business card</h1>
      <div className="businessCard-cards">
        <div className="businessCard-cardBack" />
        <div className="businessCard-cardFront">
          <div>
            <p className="businessCard-cardFront-title">{fullName}</p>
            <p className="businessCard-cardFront-subtitle">{jobDescription}</p>
          </div>
          <div className="businessCard-cardFront-bottom">
            <p className="businessCard-icon-phone">
              {prefix} {phoneNumber}
            </p>
            <p className="businessCard-icon-email">{email}</p>
            <p className="businessCard-icon-website">{website}</p>
            <p className="businessCard-icon-address">{address}</p>
          </div>
        </div>
      </div>
    </article>
  );
};

export default Card;
