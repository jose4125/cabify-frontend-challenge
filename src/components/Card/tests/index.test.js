import React from 'react';
import { shallow } from 'enzyme';

import Card from '../index';

describe('card', () => {
  const props = {
    fullName: '',
    jobDescription: '',
    prefix: '+93',
    phoneNumber: '',
    email: '',
    website: '',
    address: ''
  };

  const wrapper = shallow(<Card info={props} />);
  it('should render with out crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a cabify logo', () => {
    expect(wrapper.find('.businessCard-badge-logo').exists()).toBe(true);
  });

  it('should have a title', () => {
    expect(wrapper.find('.title-main').exists()).toBe(true);
  });

  it('should the title be a h1', () => {
    expect(wrapper.find('.title-main').type()).toEqual('h1');
  });

  it('should the title have a copy', () => {
    expect(wrapper.find('.title-main').text().length).toBeGreaterThan(0);
  });

  it('should have an empty full name', () => {
    expect(wrapper.find('.businessCard-cardFront-title').text()).toBe('');
  });

  it('should have an empty subtitle', () => {
    expect(wrapper.find('.businessCard-cardFront-subtitle').text()).toBe('');
  });

  it('should have a businessCard phone', () => {
    expect(wrapper.find('.businessCard-icon-phone').text()).toBe(
      `${props.prefix} ${props.phoneNumber}`
    );
  });

  it('should have an empty email', () => {
    expect(wrapper.find('.businessCard-icon-email').text()).toBe('');
  });

  it('should have an empty website', () => {
    expect(wrapper.find('.businessCard-icon-website').text()).toBe('');
  });

  it('should have an empty address', () => {
    expect(wrapper.find('.businessCard-icon-address').text()).toBe('');
  });

  describe('card with props values', () => {
    const props = {
      fullName: 'jhon doe',
      jobDescription: 'backend',
      prefix: '+57',
      phoneNumber: '3192345687',
      email: 'jhondoe@test.com',
      website: 'jhondoe.com',
      address: 'jhondoe street'
    };

    const wrapper = shallow(<Card info={props} />);

    it('should render a full name', () => {
      expect(wrapper.find('.businessCard-cardFront-title').text()).toBe(
        `${props.fullName}`
      );
    });

    it('should render a subtitle', () => {
      expect(wrapper.find('.businessCard-cardFront-subtitle').text()).toBe(
        `${props.jobDescription}`
      );
    });

    it('should render a businessCard phone', () => {
      expect(wrapper.find('.businessCard-icon-phone').text()).toBe(
        `${props.prefix} ${props.phoneNumber}`
      );
    });

    it('should render an email', () => {
      expect(wrapper.find('.businessCard-icon-email').text()).toBe(
        `${props.email}`
      );
    });

    it('should render a website', () => {
      expect(wrapper.find('.businessCard-icon-website').text()).toBe(
        `${props.website}`
      );
    });

    it('should render a address', () => {
      expect(wrapper.find('.businessCard-icon-address').text()).toBe(
        `${props.address}`
      );
    });
  });
});
