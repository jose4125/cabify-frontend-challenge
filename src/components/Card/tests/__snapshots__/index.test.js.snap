// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should render with out crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    info={
      Object {
        "address": "",
        "email": "",
        "fullName": "",
        "jobDescription": "",
        "phoneNumber": "",
        "prefix": "+93",
        "website": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <figure
          className="businessCard-badge"
        >
          <a
            className="businessCard-badge-logo"
            href="http://www.cabify.com"
          >
            <img
              alt="Cabify"
              src="cabify-logo.svg"
            />
          </a>
        </figure>,
        <h1
          className="title-main"
        >
          Request your business card
        </h1>,
        <div
          className="businessCard-cards"
        >
          <div
            className="businessCard-cardBack"
          />
          <div
            className="businessCard-cardFront"
          >
            <div>
              <p
                className="businessCard-cardFront-title"
              >
                
              </p>
              <p
                className="businessCard-cardFront-subtitle"
              >
                
              </p>
            </div>
            <div
              className="businessCard-cardFront-bottom"
            >
              <p
                className="businessCard-icon-phone"
              >
                +93
                 
                
              </p>
              <p
                className="businessCard-icon-email"
              >
                
              </p>
              <p
                className="businessCard-icon-website"
              >
                
              </p>
              <p
                className="businessCard-icon-address"
              >
                
              </p>
            </div>
          </div>
        </div>,
      ],
      "className": "businessCard col col6",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="businessCard-badge-logo"
            href="http://www.cabify.com"
          >
            <img
              alt="Cabify"
              src="cabify-logo.svg"
            />
          </a>,
          "className": "businessCard-badge",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Cabify"
              src="cabify-logo.svg"
            />,
            "className": "businessCard-badge-logo",
            "href": "http://www.cabify.com",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Cabify",
              "src": "cabify-logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        "type": "figure",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Request your business card",
          "className": "title-main",
        },
        "ref": null,
        "rendered": "Request your business card",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="businessCard-cardBack"
            />,
            <div
              className="businessCard-cardFront"
            >
              <div>
                <p
                  className="businessCard-cardFront-title"
                >
                  
                </p>
                <p
                  className="businessCard-cardFront-subtitle"
                >
                  
                </p>
              </div>
              <div
                className="businessCard-cardFront-bottom"
              >
                <p
                  className="businessCard-icon-phone"
                >
                  +93
                   
                  
                </p>
                <p
                  className="businessCard-icon-email"
                >
                  
                </p>
                <p
                  className="businessCard-icon-website"
                >
                  
                </p>
                <p
                  className="businessCard-icon-address"
                >
                  
                </p>
              </div>
            </div>,
          ],
          "className": "businessCard-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "businessCard-cardBack",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <p
                    className="businessCard-cardFront-title"
                  >
                    
                  </p>
                  <p
                    className="businessCard-cardFront-subtitle"
                  >
                    
                  </p>
                </div>,
                <div
                  className="businessCard-cardFront-bottom"
                >
                  <p
                    className="businessCard-icon-phone"
                  >
                    +93
                     
                    
                  </p>
                  <p
                    className="businessCard-icon-email"
                  >
                    
                  </p>
                  <p
                    className="businessCard-icon-website"
                  >
                    
                  </p>
                  <p
                    className="businessCard-icon-address"
                  >
                    
                  </p>
                </div>,
              ],
              "className": "businessCard-cardFront",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="businessCard-cardFront-title"
                    >
                      
                    </p>,
                    <p
                      className="businessCard-cardFront-subtitle"
                    >
                      
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "businessCard-cardFront-title",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "businessCard-cardFront-subtitle",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="businessCard-icon-phone"
                    >
                      +93
                       
                      
                    </p>,
                    <p
                      className="businessCard-icon-email"
                    >
                      
                    </p>,
                    <p
                      className="businessCard-icon-website"
                    >
                      
                    </p>,
                    <p
                      className="businessCard-icon-address"
                    >
                      
                    </p>,
                  ],
                  "className": "businessCard-cardFront-bottom",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "+93",
                        " ",
                        "",
                      ],
                      "className": "businessCard-icon-phone",
                    },
                    "ref": null,
                    "rendered": Array [
                      "+93",
                      " ",
                      "",
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "businessCard-icon-email",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "businessCard-icon-website",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "businessCard-icon-address",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <figure
            className="businessCard-badge"
          >
            <a
              className="businessCard-badge-logo"
              href="http://www.cabify.com"
            >
              <img
                alt="Cabify"
                src="cabify-logo.svg"
              />
            </a>
          </figure>,
          <h1
            className="title-main"
          >
            Request your business card
          </h1>,
          <div
            className="businessCard-cards"
          >
            <div
              className="businessCard-cardBack"
            />
            <div
              className="businessCard-cardFront"
            >
              <div>
                <p
                  className="businessCard-cardFront-title"
                >
                  
                </p>
                <p
                  className="businessCard-cardFront-subtitle"
                >
                  
                </p>
              </div>
              <div
                className="businessCard-cardFront-bottom"
              >
                <p
                  className="businessCard-icon-phone"
                >
                  +93
                   
                  
                </p>
                <p
                  className="businessCard-icon-email"
                >
                  
                </p>
                <p
                  className="businessCard-icon-website"
                >
                  
                </p>
                <p
                  className="businessCard-icon-address"
                >
                  
                </p>
              </div>
            </div>
          </div>,
        ],
        "className": "businessCard col col6",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="businessCard-badge-logo"
              href="http://www.cabify.com"
            >
              <img
                alt="Cabify"
                src="cabify-logo.svg"
              />
            </a>,
            "className": "businessCard-badge",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Cabify"
                src="cabify-logo.svg"
              />,
              "className": "businessCard-badge-logo",
              "href": "http://www.cabify.com",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Cabify",
                "src": "cabify-logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          "type": "figure",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Request your business card",
            "className": "title-main",
          },
          "ref": null,
          "rendered": "Request your business card",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="businessCard-cardBack"
              />,
              <div
                className="businessCard-cardFront"
              >
                <div>
                  <p
                    className="businessCard-cardFront-title"
                  >
                    
                  </p>
                  <p
                    className="businessCard-cardFront-subtitle"
                  >
                    
                  </p>
                </div>
                <div
                  className="businessCard-cardFront-bottom"
                >
                  <p
                    className="businessCard-icon-phone"
                  >
                    +93
                     
                    
                  </p>
                  <p
                    className="businessCard-icon-email"
                  >
                    
                  </p>
                  <p
                    className="businessCard-icon-website"
                  >
                    
                  </p>
                  <p
                    className="businessCard-icon-address"
                  >
                    
                  </p>
                </div>
              </div>,
            ],
            "className": "businessCard-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "businessCard-cardBack",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <p
                      className="businessCard-cardFront-title"
                    >
                      
                    </p>
                    <p
                      className="businessCard-cardFront-subtitle"
                    >
                      
                    </p>
                  </div>,
                  <div
                    className="businessCard-cardFront-bottom"
                  >
                    <p
                      className="businessCard-icon-phone"
                    >
                      +93
                       
                      
                    </p>
                    <p
                      className="businessCard-icon-email"
                    >
                      
                    </p>
                    <p
                      className="businessCard-icon-website"
                    >
                      
                    </p>
                    <p
                      className="businessCard-icon-address"
                    >
                      
                    </p>
                  </div>,
                ],
                "className": "businessCard-cardFront",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="businessCard-cardFront-title"
                      >
                        
                      </p>,
                      <p
                        className="businessCard-cardFront-subtitle"
                      >
                        
                      </p>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "businessCard-cardFront-title",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "businessCard-cardFront-subtitle",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="businessCard-icon-phone"
                      >
                        +93
                         
                        
                      </p>,
                      <p
                        className="businessCard-icon-email"
                      >
                        
                      </p>,
                      <p
                        className="businessCard-icon-website"
                      >
                        
                      </p>,
                      <p
                        className="businessCard-icon-address"
                      >
                        
                      </p>,
                    ],
                    "className": "businessCard-cardFront-bottom",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "+93",
                          " ",
                          "",
                        ],
                        "className": "businessCard-icon-phone",
                      },
                      "ref": null,
                      "rendered": Array [
                        "+93",
                        " ",
                        "",
                      ],
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "businessCard-icon-email",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "businessCard-icon-website",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "",
                        "className": "businessCard-icon-address",
                      },
                      "ref": null,
                      "rendered": "",
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
